pointerevents/pointerevent_mouse_pointercapture_inactivate_pointer.html WPT is failing

nt1m opened the pull request:
#### 3ea46f4e0b98e3db66ae076e75273d059de8da4d
<pre>
pointerevents/pointerevent_mouse_pointercapture_inactivate_pointer.html WPT is failing
<a href="https://bugs.webkit.org/show_bug.cgi?id=274640">https://bugs.webkit.org/show_bug.cgi?id=274640</a>
<a href="https://rdar.apple.com/128669314">rdar://128669314</a>

Reviewed by Antoine Quint.

Implement a missing bit of the pointer events spec where we weren&apos;t checking against the &quot;active document&quot; of the pointer when setting the pointer capture.

<a href="https://w3c.github.io/pointerevents/#dfn-set-pointer-capture">https://w3c.github.io/pointerevents/#dfn-set-pointer-capture</a>

* LayoutTests/imported/w3c/web-platform-tests/pointerevents/pointerevent_mouse_pointercapture_inactivate_pointer-expected.txt:
* LayoutTests/platform/mac/TestExpectations:
Rebaseline and unskip test.

* Source/WebCore/page/PointerCaptureController.cpp:
(WebCore::PointerCaptureController::setPointerCapture):
Implement <a href="https://w3c.github.io/pointerevents/#ref-for-dfn-active-document-2">https://w3c.github.io/pointerevents/#ref-for-dfn-active-document-2</a>

(WebCore::PointerCaptureController::pointerEventWillBeDispatched):
Implement <a href="https://w3c.github.io/pointerevents/#ref-for-dfn-active-document-1">https://w3c.github.io/pointerevents/#ref-for-dfn-active-document-1</a>

* Source/WebCore/page/PointerCaptureController.h:

Store active document in CapturingData: <a href="https://w3c.github.io/pointerevents/#dfn-active-document">https://w3c.github.io/pointerevents/#dfn-active-document</a>
Canonical link: <a href="https://commits.webkit.org/286098@main">https://commits.webkit.org/286098@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5ec0da5b363798d1edc02a3e75a88c32961c943e

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/74843 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/54273 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/27660 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/79268 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/26081 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/76960 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/63409 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/2058 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/79268 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/17064 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/77910 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/48938 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/64310 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/79268 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/46217 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/21807 "Passed tests") | [  ~~🛠 wpe-cairo~~](https://ews-build.webkit.org/#/builders/65/builds/24414 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/67382 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/22152 "Passed tests") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/80757 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 vision~~](https://ews-build.webkit.org/#/builders/128/builds/2161 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/1300 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/80757 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 vision-sim~~](https://ews-build.webkit.org/#/builders/121/builds/2309 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/64328 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/80757 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/16474 "Built successfully and passed tests") | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-2-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/119/builds/8427 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/2126 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/4913 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/125/builds/2154 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/129/builds/3075 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/124/builds/2161 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

graouts commented on Source/WebCore/page/PointerCaptureController.h:
```diff
@@ -90,6 +90,7 @@ class PointerCaptureController {
                 ;
         }
         String pointerType;
+        RefPtr<Document> activeDocument;
```
Should this be a `WeakPtr` instead?

nt1m replied:
Good suggestion, pointer events shouldn't hold the document alive, I'll address this after EWS is done running tests.

graouts reviewed (approved):
I don't follow the changes in `LayoutTests/platform/Mac/TestExpectations`. How are those related to the source change?

nt1m committed:
pointerevents/pointerevent_mouse_pointercapture_inactivate_pointer.html WPT is failing
https://bugs.webkit.org/show_bug.cgi?id=274640
rdar://128669314

Reviewed by Antoine Quint.

Implement a missing bit of the pointer events spec where we weren't checking against the "active document" of the pointer when setting the pointer capture.

https://w3c.github.io/pointerevents/#dfn-set-pointer-capture

* LayoutTests/imported/w3c/web-platform-tests/pointerevents/pointerevent_mouse_pointercapture_inactivate_pointer-expected.txt:
* LayoutTests/platform/mac/TestExpectations:
Rebaseline and unskip test.

* Source/WebCore/page/PointerCaptureController.cpp:
(WebCore::PointerCaptureController::setPointerCapture):
Implement https://w3c.github.io/pointerevents/#ref-for-dfn-active-document-2

(WebCore::PointerCaptureController::pointerEventWillBeDispatched):
Implement https://w3c.github.io/pointerevents/#ref-for-dfn-active-document-1

* Source/WebCore/page/PointerCaptureController.h:

Store active document in CapturingData: https://w3c.github.io/pointerevents/#dfn-active-document
Canonical link: https://commits.webkit.org/286098@main
